bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/handler/MieleBridgeHandler.java

```diff
@@ -308,117 +316,148 @@ private List<HomeDevice> getHomeDevices() throws MieleRpcException {
+                        }
+                        var deviceProperty = new DeviceProperty();
+                        deviceProperty.Name = name;
+                        deviceProperty.Value = value;
+                        ApplianceStatusListener listener = applianceStatusListeners
+                                .get(applianceIdentifier.getApplianceId());
+                        if (listener != null) {
+                            listener.onAppliancePropertyChanged(deviceProperty);
+                        }
+                    } catch (SocketTimeoutException e) {
+                        try {
+                            Thread.sleep(MULTICAST_SLEEP_MILLIS);
+                        } catch (InterruptedException ex) {
+                            Thread.currentThread().interrupt();
+                            logger.debug("Event listener has been interrupted.");
+                            break;
                         }
-                    } catch (IOException e) {
-                        logger.debug("An exception occurred while leaving multicast group: '{}'", e.getMessage());
                     }
+                }
+            } catch (IOException e) {
+                logger.debug("An exception occurred while receiving multicast packets: '{}'", e.getMessage());
+            } finally {
+                // restart the cycle with a clean slate
+                try {
                     if (clientSocket != null) {
-                        clientSocket.close();
+                        clientSocket.leaveGroup(address1, null);
+                        clientSocket.leaveGroup(address2, null);
                     }
+                } catch (IOException e) {
+                    logger.debug("An exception occurred while leaving multicast group: '{}'", e.getMessage());
+                }
+                if (clientSocket != null) {
+                    clientSocket.close();
                 }
             }
-        } else {
-            logger.debug("Invalid IP address for the multicast interface: '{}'", getConfig().get(INTERFACE));
         }
     };
 
+    private @Nullable NetworkInterface getMulticastInterface(String interfaceIpAddress) throws SocketException {
+        Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+
+        @Nullable
+        NetworkInterface networkInterface;
+        while (networkInterfaces.hasMoreElements()) {
+            networkInterface = networkInterfaces.nextElement();
+            if (networkInterface.isLoopback()) {
+                continue;
+            }
+            for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                if (logger.isTraceEnabled()) {
+                    logger.trace("Found interface address {} -> {}", interfaceAddress.toString(),
+                            interfaceAddress.getAddress().toString());
+                }
```

Comment:
Is it really useful to ckeck if TRACE is enabled ?

Reply:
I wanted for performance reasons to prevent three usually unneeded method calls: `toString()`, `getAddress()` and `toString()`.

Reply:
interfaceAddress.getAddress().toString() is called just after and you could have used a variable, but ok for interfaceAddress.toString()

Reply:
Oh right, I guess this slipped in during refactoring. So there will be a slight performance hit when having trace logging enabled. I guess it's acceptable, but I will fix this next time I'm touching code in this area. 🙂